src/System.Net.Security/src/System/Net/SecureChannel.cs

```diff
@@ -1059,16 +1060,12 @@ internal SecurityStatusPal Encrypt(byte[] buffer, int offset, int size, ref byte
                 resultSize = 0;
 
                 int bufferSizeNeeded = checked(size + _headerSize + _trailerSize);
-                if (output != null && bufferSizeNeeded <= output.Length)
-                {
-                    writeBuffer = output;
-                }
-                else
+
```

Comment:
Consider moving the calculation of bufferSizeNeeded to behind the SslStreamPal.PresizeEncryptBuffer guard.